### PR TITLE
Add CLI env variables for `dev`, `build` and `serve` commands

### LINKED ISSUE #394: Add ENV variables for the build, dev and serve commands. [3pt]
**Feature:** [Change publicPath](https://community.frontity.org/t/change-publicpath/1461).
**User Story**: Change the publicPath at build time.

[**Sprint Card**](https://github.com/orgs/frontity/projects/4#card-36476088)

### PR DESCRIPTION
**Pull request by DAreRodz**
**What**:

Add these env variables to the `dev`, `build` and `serve` commands:

**`dev`**

- target: `FRONTITY_DEV_TARGET`
- port: `FRONTITY_DEV_PORT`
- https: `FRONTITY_DEV_HTTPS`
- production: `FRONTITY_DEV_PRODUCTION`
- publicPath: `FRONTITY_DEV_PUBLIC_PATH`
- dont-open-browser: `FRONTITY_DEV_DONT_OPEN_BROWSER`

**`build`**

- target: `FRONTITY_BUILD_TARGET`
- development: `FRONTITY_BUILD_DEVELOPMENT`
- publicPath: `FRONTITY_BUILD_PUBLIC_PATH`

**`serve`**

- port: `FRONTITY_SERVE_PORT`
- https: `FRONTITY_SERVE_HTTPS`

Closes #394 

**Why**:

To allow users to pass CLI options through env variables.

**How**:

Following the implementation proposal in [Change publicPath](https://community.frontity.org/t/change-publicpath/1461/5).

**Tasks**:

- [x] Code
- [x] Unit tests
- [x] Community discussions: https://community.frontity.org/t/change-publicpath/1461/8
- [x] Changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)
- [X] Link to PR in [Documentation](https://github.com/frontity/gitbook-docs/) → https://github.com/frontity/gitbook-docs/pull/93

**Unrelated Tasks**

- TSDocs
- TypeScript
- End to end tests
- TypeScript tests
- Update starter themes
- Update other packages

### CODE REVIEW

**Comment by luisherranz** [comment]
@luisherranz test

**Comment by DAreRodz** [comment]
By the way, Codecov is failing because, after adding unit tests for `dev`, `build` and `serve` commands, it started to track files that were not tracked before, thus causing the coverage ratio to drop.

I tried to fix that writing some missing unit tests for other parts of the code but it wasn't enough, and I think that work if out of the scope of this PR, so I would merge this at it is.

@michalczaplinski, do you know what would happen with the coverage ratio if we merge this? Would it be reset to the score that this PR is getting?

**Comment by DAreRodz** [comment]
@juanmaguitar, there is no TSdocs in this case.

You would have to create a new section for each command called **Environment variables** or something like that, or add them to their respective argument.

Whatever you think is better. :+1:

**Comment by michalczaplinski** [comment]
@DAreRodz yes, I'm afraid that the codecov is going to decrease by 0.95% after merging this. 

The thing is that this does not mean that our code is less tested than before actually...

In fact, if the file was not tracked before, but is tracked now and the new test only covers 10% of the new file, the coverage is going to decrease even though we actually test MORE of the functionality.

I think we have to look into configuring jest correctly - there exist multiple additional flags for configuring coverage that for some reason are not present on the website: https://github.com/facebook/jest/blob/master/docs/Configuration.md

**Comment by luisherranz** [comment]
Hey guys, I have a concern about this environment variables names clashing in the future. Could you please take a look and let me know your opinion before I continue with the review? Thanks!

https://community.frontity.org/t/change-publicpath/1461/9?u=luisherranz

I am moving this to blocked until then.

**Comment by luisherranz** [comment]
Unblocked based on the decision made at https://community.frontity.org/t/change-publicpath/1461/13.

**Review by luisherranz — CHANGES_REQUESTED** [review]
Unblocked based on the decision made at https://community.frontity.org/t/change-publicpath/1461/13.

@DAreRodz could you please implement the necessary changes?

**Review by juanmaguitar — APPROVED** [review]
📕 Docs PR → https://github.com/frontity/gitbook-docs/pull/93

File: packages/frontity/src/cli/index.ts
Comment (luisherranz) [inline packages/frontity/src/cli/index.ts:92]: Thanks for this fix 🙂

**Review by luisherranz — APPROVED** [review]
Thanks @darerodz! Looks perfect to me 🙂

**Pull request merged**
